__tests__/unit-tests/FriendScreen-test.tsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+import { render, fireEvent, waitFor } from "@testing-library/react-native";
+import FriendsScreen from "@/app/screens/home/friends_screen";
+import { ThemedView } from "@/components/theme/ThemedView";
+
+// Mock des dépendances
+jest.mock("@/components/theme/ThemedView", () => ({
+  ThemedView: ({ children }: any) => <div>{children}</div>,
+}));
+
+jest.mock("@/components/theme/ThemedText", () => ({
+  ThemedText: ({ children }: any) => <span>{children}</span>,
+}));
+
+jest.mock("@/components/friends/ListOfFilteredUsers", () => ({
+  __esModule: true,
+  default: ({ searchText }: any) => (
+    <div data-testid="filtered-users">{searchText}</div>
+  ),
+}));
+
+jest.mock("@/components/friends/ListOfFriends", () => ({
+  __esModule: true,
+  default: ({ friends }: any) => (
+    <div data-testid="friends-list">
+      {friends.map((friend: any) => friend.name).join(", ")}
+    </div>
+  ),
+}));
+
+jest.mock("@/components/friends/RequestsList", () => ({
+  __esModule: true,
+  default: (requests: any) => (
+    <div data-testid="requests-list">
+      {requests.map((request: any) => request.name).join(", ")}
+    </div>
+  ),
+}));
+
+const mockFirestoreCtrl = {
+  getAllUsers: jest.fn(),
+  getFriends: jest.fn(),
+  getFriendRequests: jest.fn(),
+};
+
+const mockNavigation = {
+  goBack: jest.fn(),
+  navigate: jest.fn(),
+};
+
+describe("FriendsScreen", () => {
+  const mockUID = "mock-uid";
+  const mockAuth = {
+    currentUser: { uid: mockUID },
+  };
+  jest.mock("firebase/auth", () => ({
+    getAuth: () => mockAuth,
```

Comment:
If using getUser from the FirestoreCtrl class you wouldn't have to mock it due to it already being mocked in the jestSetupFile

Reply:
Good catch thanks !